### PR TITLE
feat: Adding allow inspecting inner fields of nw.Struct

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Related issue # 
- Closes #https://github.com/narwhals-dev/narwhals/issues/1107

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

### CODE REVIEW

File: narwhals/_pandas_like/utils.py [hunk 1]
```diff
@@ -294,7 +294,7 @@ def native_to_narwhals_dtype(native_column: Any, dtypes: DTypes) -> DType:
             native_column.dtype.pyarrow_dtype.list_size,
         )
     if dtype.startswith("struct"):
-        return dtypes.Struct()
+        return arrow_native_to_narwhals_dtype(native_column.dtype.pyarrow_dtype, dtypes)
```
Comment (MarcoGorelli) [inline narwhals/_pandas_like/utils.py:297]: this gets me thinking, can the `if dtype.startswith(("large_list", "list")):` branch above be simplified to also just return this? (for a separate pr)

File: tests/frame/schema_test.py [hunk 2]
```diff
@@ -222,7 +221,7 @@ def test_nested_dtypes() -> None:
     nwdf = nw.from_native(df)
     assert nwdf.schema == {"a": nw.List, "b": nw.Array(nw.Int64, 2), "c": nw.Struct}
     df = pl.DataFrame(
-        {"a": [[1, 2]], "b": [[1, 2]], "c": [{"a": 1}]},
+        {"a": [[1, 2]], "b": [[1, 2]], "c": [{"a": 1, "b": "x", "c": 1.1}]},
```
Comment (MarcoGorelli) [inline tests/frame/schema_test.py:224]: at the end of this test, there's
```python
assert nwdf.schema == {"a": nw.List, "b": nw.Array(nw.Int64, 2), "c": nw.Struct}
```
maybe we should make it
```python
assert nwdf.schema == {"a": nw.List, "b": nw.Array(nw.Int64, 2), "c": nw.Struct}
assert nwdf.schema == {"a": nw.List(nw.Int64), "b": nw.Array(nw.Int64, 2), "c": nw.Struct({'a': nw.Int64, 'b': nw.String, 'c': nw.Float64}}
```
? and similarly for the others here in this file, just to double-check that we're really covering everything (and doing so across the different backends)

**Review by MarcoGorelli — COMMENTED** [review]
wow, this is literal 🔥

well done, this pleases me to no end 🤗 

---

i'll take a look at the downstream test - it's "just" typing though so hopefully there's a simple resolution

File: narwhals/_pandas_like/utils.py (hunk 1 above)
Comment (DeaMariaLeon) [reply narwhals/_pandas_like/utils.py:297]: I'll do that.

File: tests/frame/schema_test.py (hunk 2 above)
Comment (DeaMariaLeon) [reply tests/frame/schema_test.py:233]: Done, thanks for the suggestion.

**Comment by DeaMariaLeon** [comment]
> it's "just" typing

Typing can be nightmarish 🥹


> wow, this is literal 🔥
> 
> well done, this pleases me to no end 🤗

Now, that's feedback.. Thanks! 😊

**Review by MarcoGorelli — APPROVED** [review]
awesome work, thanks @DeaMariaLeon !

**Pull request merged**